### PR TITLE
Use resolve_length in ShapeBuilder.rect

### PR DESCRIPTION
**Pull request by darioackermann**
Provides a tiny fix to #555  

**Checklist**:

- [ ] The GitHub pipeline is OK (green), 
      meaning that both `pylint` (static code analyzer) and `black` (code formatter) are happy with the changes of this PR.

- [x] A unit test is covering the code added / modified by this PR

- [ ] This PR is ready to be merged 

- [x] In case of a new feature, docstrings have been added, with also some documentation in the `docs/` folder

- [x] A mention of the change is present in `CHANGELOG.md`

By submitting this pull request, I confirm that my contribution is made under the terms of the [GNU LGPL 3.0 license](https://github.com/PyFPDF/fpdf2/blob/master/LICENSE).

### CODE REVIEW

**Comment by Lucas-C** [comment]
Thank you for submitting a PR!
Would you mind adding a mention of this fix in `CHANGELOG.md`, and also a small unit test please?
You can just add an entry here: https://github.com/PyFPDF/fpdf2/blob/master/test/svg/parameters.py#L73

**Comment by darioackermann** [comment]
Added a Test and documented in Changelog. Should there any change be necessary, just let me know.

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ This can also be enabled programmatically with `warnings.simplefilter('default',
 - automatic page break is never performed on an empty page (when the Y position is at the top margin)
 - fixed [`insert_toc_placeholder()`](https://pyfpdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.insert_toc_placeholder) usage with [`footer()`](https://pyfpdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.footer) and `{{nb}}`; [#548](https://github.com/PyFPDF/fpdf2/issues/548)
 - the SVG parser now accepts stroke-width attribute values with an explicit unit, thanks to @gmischler; [#526](https://github.com/PyFPDF/fpdf2/issues/526)
+- the SVG parser now accepts absolute units for `width` and `height` attributes, thanks to @darioackermann; [#555](https://github.com/PyFPDF/fpdf2/issues/555)
```
Comment (Lucas-C) [inline CHANGELOG.md:31]: 👍

File: test/svg/parameters.py [hunk 2]
```diff
@@ -71,6 +71,12 @@ def Gs(**kwargs):
 
 
 test_svg_shape_tags = (
+    pytest.param(
+        '<rect x="20" y="20" width="25.4mm" height="2.54 cm"/>',
```
Comment (Lucas-C) [inline test/svg/parameters.py:75]: 👍

**Review by Lucas-C — APPROVED** [review]

**Comment by Lucas-C** [comment]
Your PR is fine but the GitHub Actions pipeline is currently blocked by #559
I'll try to fix this asap and you'll just have to rebase your branch

**Comment by darioackermann** [comment]
Sure, no stress! I'll watch #559 and rebase accordingly

**Comment by Lucas-C** [comment]
This should be fixed now

**Comment by Lucas-C** [comment]
I rebased and based your branch
Thank you @darioackermann!

**Comment by darioackermann** [comment]
Thanks for rebasing, I was a bit short of time today. You're welcome!

**Pull request merged**